### PR TITLE
[fs-extra] 2.0.0 obsoletes walk and walkSync

### PR DESCRIPTION
**Pull request by shiftkey**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `tsc` without errors.
- ~~[ ] Run `npm run lint package-name` if a `tslint.json` is present.~~ 

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes

This is mentioned in [the NPM package](https://www.npmjs.com/package/fs-extra#what-happened-to-walk-and-walksync-) but here's the PR where it was obsoleted https://github.com/jprichardson/node-fs-extra/pull/339.

screen shot 2017-03-06 at 2 44 29 pm

I'll follow up with subsequent PRs to ensure the type declarations exist for those packages.

- [x] Increase the version number in the header if appropriate.
- ~~[ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.~~

### CODE REVIEW

File: fs-extra/index.d.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
-// Type definitions for fs-extra
+// Type definitions for fs-extra v2.0.0
```
Comment (shiftkey) [inline fs-extra/index.d.ts]: I'm not sure if this is the right way to version this, but I added this to ensure we're not confusing `1.x` usage with `2.x` usage.

**Comment by shiftkey** [comment]
re-introducing `walkSync` by adding new `klaw-sync` declarations: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/15005

**Comment by minestarks** [comment]
@midknight41 any feedback?

**Comment by midknight41** [comment]
👍

**Comment by midknight41** [comment]
I'm not sure on the versioning question tbh.

**Pull request merged**